### PR TITLE
Update configured plugins

### LINKED ISSUE #140: Init container "copy-default-config" fails with "cp: cannot stat '/usr/share/jenkins/ref/plugins/*': No such file or directory"
**Describe the bug**
The installation of the chart with all the default values fails to correctly install the plugins. The init container `copy-default-config` fails with:

```
kubectl logs jenkins-6579c7b576-rcm9q copy-default-config

applying Jenkins configuration
disable Setup Wizard
download plugins
Plugin workflow-aggregator:2.6 depends on workflow-job:2.40, but there is an older version defined on the top level - workflow-job:2.39
copy plugins to shared volume
cp: cannot stat '/usr/share/jenkins/ref/plugins/*': No such file or directory
finished initialization
```

**Version of Helm and Kubernetes**:

Helm Version:

```console
$ helm version
version.BuildInfo{Version:"v3.3.1", GitCommit:"249e5215cde0c3fa72e27eb7a30e8d55c9696144", GitTreeState:"clean", GoVersion:"go1.14.7"}
```

Kubernetes Version:

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Which version of the chart**:
2.17.1

**What happened**:
Jenkins is deployed without any plugins

**What you expected to happen**:
Jenkins being installed with all the provided plugins

**How to reproduce it** (as minimally and precisely as possible):
`helm install jenkins jenkins/jenkins`

**Anything else we need to know**:

It works correctly with the `2.17.0` version of the chart.

### PR DESCRIPTION
**Pull request by torstenwalter**
# What this PR does / why we need it

- Use lts version 2.249.3
- Update kubernetes, workflow-aggregator, git and configuration-as-code plugins.
- remove plugins which are provided as dependency
- Fail apply_config.sh script if an error occurs.

# Which issue this PR fixes

- fixes #140

# Special notes for your reviewer

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/Chart.yaml [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 name: jenkins
 home: https://jenkins.io/
-version: 2.18.2
-appVersion: lts
-description: Open source continuous integration server. It supports multiple SCM tools
-  including CVS, Subversion and Git. It can execute Apache Ant and Apache Maven-based
-  projects as well as arbitrary scripts.
+version: 2.19.0
+appVersion: 2.249.3
+description: The leading open source automation server, Jenkins provides hundreds of plugins to support building, deploying and automating any project.
```
Comment (timja) [inline charts/jenkins/Chart.yaml:6]: Over 1500+ plugins I think is the tag line cc @oleg-nenashev

Comment (torstenwalter) [reply charts/jenkins/Chart.yaml:6]: What about this?

```suggestion
description: Jenkins - Build great things at any scale! The leading open source automation server, Jenkins provides hundreds of plugins to support building, deploying and automating any project.
```

This is from https://www.jenkins.io/
> Jenkins
> Build great things at any scale
> 
> The leading open source automation server, Jenkins provides hundreds of plugins to support building, deploying and automating any project.
>

Comment (timja) [reply charts/jenkins/Chart.yaml:6]: sure

**Review by timja — APPROVED** [review]

**Comment by torstenwalter** [comment]
@timja I thought that issues with the mirroring infrastructure have been resolved:

```
INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://mirror.serverion.com:443: Connection reset
Nov 21, 2020 9:49:38 AM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->https://mirror.serverion.com:443
Nov 21, 2020 9:50:55 AM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.net.SocketException) caught when processing request to {s}->https://mirror.serverion.com:443: Connection reset
```

**Comment by timja** [comment]
It is, probably serverion mirror had a issue, retry it

**Pull request merged**